tools/cdn.sh

```diff
@@ -1,27 +1,42 @@
+
 CURRENT_VERSION=$(node tools/attribute.js version)
+MAJORMINOR_VERSION_ONLY=$(node tools/attribute.js version ../package.json majorminor)
 EXTENSION_NAME="auth0-account-link"
 
-BUNDLE_EXISTS=$(aws s3 ls s3://assets.us.auth0.com/extensions/$EXTENSION_NAME/ | grep "$EXTENSION_NAME-$CURRENT_VERSION.js")
-CDN_EXISTS=$(aws s3 ls s3://assets.us.auth0.com/extensions/$EXTENSION_NAME/assets/ | grep "link.$CURRENT_VERSION.min.css")
-ADMIN_CDN_EXISTS=$(aws s3 ls s3://assets.us.auth0.com/extensions/$EXTENSION_NAME/assets/ | grep "admin.$CURRENT_VERSION.min.css")
-
-if [ ! -z "$BUNDLE_EXISTS" ]; then
-  echo "There is already a $EXTENSION_NAME-$CURRENT_VERSION.js in the cdn. Skipping cdn publish…"
-else
-  aws s3 cp build/bundle.js s3://assets.us.auth0.com/extensions/$EXTENSION_NAME/$EXTENSION_NAME-$CURRENT_VERSION.js --region us-west-1 --acl public-read
-  echo "$EXTENSION_NAME-$CURRENT_VERSION.js uploaded to the cdn"
-fi
-
-if [ ! -z "$CDN_EXISTS" ]; then
-  echo "There is already a link.$CURRENT_VERSION.min.css in the cdn. Skipping cdn publish…"
-else
-  aws s3 cp dist/assets/link.$CURRENT_VERSION.min.css s3://assets.us.auth0.com/extensions/$EXTENSION_NAME/assets/link.$CURRENT_VERSION.min.css --region us-west-1 --cache-control "max-age=86400" --acl public-read
-  echo "link.$CURRENT_VERSION.min.css uploaded to the cdn"
-fi
-
-if [ ! -z "$ADMIN_CDN_EXISTS" ]; then
-  echo "There is already a admin.$CURRENT_VERSION.min.css in the cdn. Skipping cdn publish…"
-else
-  aws s3 cp dist/assets/admin.$CURRENT_VERSION.min.css s3://assets.us.auth0.com/extensions/$EXTENSION_NAME/assets/admin.$CURRENT_VERSION.min.css --region us-west-1 --cache-control "max-age=86400" --acl public-read
-  echo "admin.$CURRENT_VERSION.min.css uploaded to the cdn"
-fi
+deploy_bundle() {
+  if [ -z "$1" ]; then
+    echo "No version provided. Skipping cdn publish…"
+    exit 1
+  fi
+
+  BUNDLE_EXISTS=$(aws s3 ls s3://assets.us.auth0.com/extensions/$EXTENSION_NAME/ | grep "$EXTENSION_NAME-$1.js")
+  CDN_EXISTS=$(aws s3 ls s3://assets.us.auth0.com/extensions/$EXTENSION_NAME/assets/ | grep "link.$1.min.css")
+  ADMIN_CDN_EXISTS=$(aws s3 ls s3://assets.us.auth0.com/extensions/$EXTENSION_NAME/assets/ | grep "admin.$1.min.css")
+
+  if [ ! -z "$BUNDLE_EXISTS" ]; then
```

Comment:
I think this existence check should be skipped if we're uploading the major/minor version. Since in most cases there will already be a major/minor version we want to replace, not skip.

Reply:
Pushing up this logic
```
  if [ ! -z "$BUNDLE_EXISTS" ] && [ "$1" == "$CURRENT_VERSION" ]; then ...
```
If it's `majorminor` (2.x) it will always deploy. if its full version (2.3.x) it'll check if the bundle exists.